orangecontrib/spectroscopy/data.py

```diff
@@ -1014,16 +1024,118 @@ def read_v2(self):
+
+
+        data_gsf_a = self._gsf_reader(file_gsf_a)
+        data_gsf_p = self._gsf_reader(file_gsf_p)
+        info = self._html_reader(file_html)
+
+        final_data, parameters, final_metas = self._format_file(data_gsf_a, data_gsf_p, info)
+
+
+        metas = [Orange.data.ContinuousVariable.make("run"),
+                Orange.data.ContinuousVariable.make("row"),
+                Orange.data.ContinuousVariable.make("column"),
+                Orange.data.StringVariable.make("channel")]
+
+        domain = Orange.data.Domain([], None, metas=metas)
+        meta_data = Table.from_numpy(domain, X=np.zeros((len(final_data), 0)),
+                                     metas=np.asarray(final_metas, dtype=object))
+        
+        meta_data.attributes = parameters
+
+        depth = np.arange(0, int(parameters['Pixel Area (X, Y, Z)'][3]))
+
+        return depth, final_data, meta_data
+
+    def _format_file(self, gsf_a, gsf_p, parameters):
+        
+        info = {}
+        for row in range(len(parameters)):
+            info.update({parameters[row,0].replace(':', ''): parameters[row,1:]})
+
+        averaging = int(info['Averaging'][1])
+        px_x = int(info['Pixel Area (X, Y, Z)'][1]) # 10
+        px_y = int(info['Pixel Area (X, Y, Z)'][2]) # 10
+        px_z = number_of_points = int(info['Pixel Area (X, Y, Z)'][3]) # 1024
+        description = info['Description'][1]
+
+        data_complete = []
+        final_metas = []
+        for y in range(0, px_y):
+            amplitude = gsf_a[y].reshape((1, px_x * px_z * averaging))[0]
+            phase = gsf_p[y].reshape((1, px_x * px_z * averaging))[0]
+            i = 0
+            f = i + px_z
+            for x in range(0, px_x):
+                for run in range(0, averaging):
+                    data_complete += [amplitude[i:f]]
+                    data_complete += [phase[i:f]]
+                    
+                    final_metas += [[run,x, y, self.channel_a]]
+                    final_metas += [[run,x, y, self.channel_p]]
+                    i = f
+                    f = i + px_z
+
+        return np.asarray(data_complete), info, final_metas
+
+    def _html_reader(self, path):
+        from pandas import read_html
+        try:
+            parameters = np.asarray(read_html(path)[0])
+        except ImportError:
+            print('''Probably you need to install xlml\nTry\n "pip install lxml"\nor\n"conda install -c conda-forge lxml"''')
+            pass
+
+        return parameters
+
+    
+    def _gsf_reader(self, path):
+
+        gsf = GSFReader(path)
+        X = gsf.read_spectra()
+
+        return np.asarray(X)
+
+
     def read_spectra(self):
         version = 1
-        with open(self.filename, "rt") as f:
-            if f.read(2) == '# ':
-                version = 2
+        try:
+            with open(self.filename, "rt") as f:
+                if f.read(2) == '# ':
+                    version = 2
+        except:
+            pass
+        try:
```

Comment:
Empty excepts always look suspicious. As I advised in another comment, if you make your new functionality in a separate class you will avoid them. Simpler structure = more stable problemas.